projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-selection.service.ts

```diff
@@ -160,15 +159,16 @@ export class IgxTreeGridSelectionService extends IgxGridSelectionService {
      * @param visibleRowIDs list of all visible rowIds
      * @returns a new set with all direct parents of the rows within rowsToBeProcessed set
      */
-    private collectRowsChildrenAndDirectParents(rowsToBeProcessed: Set<any>, visibleRowIDs: Set<any>, adding: boolean): Set<any> {
+    private collectRowsChildrenAndDirectParents(rowsToBeProcessed: Set<any>, visibleRowIDs: Set<any>, adding: boolean, shouldConvert: boolean): Set<any> {
         const processedRowsParents = new Set<any>();
-        Array.from(rowsToBeProcessed).forEach((rowID) => {
+        Array.from(rowsToBeProcessed).forEach((row) => {
+            const rowID = shouldConvert ? row[this.grid.primaryKey] : row;
```

Comment:
if PK is not set this result in rowID === undefined, which is then added to selection Set and possibly causing bugs. SHould be fixed.